code/modules/admin/view_variables/debug_variable_appearance.dm

```diff
@@ -0,0 +1,219 @@
+	. = used_variables
+	var/list/unused_var_names = list()
+
+	var/image/dummy_image = image(null, null)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, appearance) // it only does self-reference
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, x) // xyz are always 0
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, y)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, z)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, weak_reference) // it's not a good idea to make a weak_ref on this, and this won't have it
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, vars) // inherited from /image, but /appearance hasn't this
+
+	// Even if these vars are essential for image, these only exists in an actual type
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, filter_data)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, realized_overlays)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, realized_underlays)
+
+	// we have no reason to show these, right?
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, _active_timers)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, _datum_components)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, _listen_lookup)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, _signal_procs)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, __auxtools_weakref_id)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, _status_traits)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, cooldowns)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, datum_flags)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, verbs)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, gc_destroyed)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, harddel_deets_dumped)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, open_uis)
+	ADD_UNUSED_VAR(unused_var_names, dummy_image, tgui_shared_states)
+
+	for(var/each in dummy_image.vars) // try to inherit var list from /image
+		if(each in unused_var_names)
+			continue
+		used_variables += each
+	del(dummy_image)
+	dummy_image = null
+
+	return used_variables
+
+/// debug_variable() proc but made for /appearance type specifically
+/proc/debug_variable_appearance(var_name, appearance)
+	var/value
+	try
+		value = locate_appearance_variable(var_name, appearance)
+	catch
+		return "<li style='backgroundColor:white'>(READ ONLY) <font color='blue'>[var_name] = (untrackable)</font></li>"
+	if(value == RESULT_VARIABLE_NOT_FOUND)
+		return "<li style='backgroundColor:white'>(READ ONLY) [var_name] <font color='blue'>(Undefined var name in switch)</font></li>"
+	return "<li style='backgroundColor:white'>(READ ONLY) [var_name] = [_debug_variable_value(var_name, value, 0, appearance, sanitize = TRUE, display_flags = NONE)]</li>"
+
+/// manually locate a variable through string value.
+/// appearance type needs a manual var referencing because it doesn't have "vars" variable internally.
+/// There's no way doing this in a fancier way.
+/proc/locate_appearance_variable(var_name, atom/movable/appearance) // it isn't /movable. It had to be at it to use NAMEOF macro
```

Comment:
sorry for keeping you in limbo for a bit, got busy and was puzzled after I saw some snippets from the Beestation review (specifically the one about aliasing for `/atom/movable` here. i think i've figured out something that might work though, `parent_type`.

basically what we do is we do something like `client_interface` where we have a representative datum to "mock" out the potential extant vars on an appearance and access that instead of having the confusing `movable` stuff be involved here. what we would do is just

```dm
/// An alias datum that allows us to access and view the variables of an appearance by keeping certain known, yet undocumented, variables that we can access and read in a datum for debugging purposes.
/// Kindly do not use this outside of a debugging context.
/datum/appearance_data
 parent_type = /atom/movable // This is necessary to access the variables on compile-time.
```

Then, we can also potentially resolve some of the OD issues like `NAMEOF()` and `mouse_drop_zone` by declaring it as a var on this datum. It's a really hacky alias technique that came to me in a dream a few nights ago. Even if we can't resolve the `NAMEOF()` stuff not working, it'll be slightly better than what we're presently doing instead of treating an appearance like a movable atom. Just test this out for me and it'll be good to ship.

Reply:
That's actually a good idea, but note that I'll do that differently in a few points:
* `for(var/each in dummy_image.vars)` is bad here because it will copy all atom/movable vars and we need to remove manually a lot.
* I'd do `/image/appearance` instead of `/datum/appearance_data` to mock more obvious.

Reply:
That's fine